apis/v1alpha2/nginxproxy_types.go

```diff
@@ -297,6 +297,17 @@ type NginxLogging struct {
 	// +optional
 	// +kubebuilder:default=info
 	AgentLevel *AgentLogLevel `json:"agentLevel,omitempty"`
+
+	// LogFormats defines custom log formats that can be used in access logs.
+	// Each log format must have a unique name.
+	//
+	// +optional
+	LogFormats []LogFormat `json:"logFormats,omitempty"`
+
+	// AccessLogs defines the access log settings, including the log file path, format, and optional parameters.
```

Comment:
While `log_format` can only be specified in the `http` context`, `access_log` can be specified at pretty much any context. This can be useful for users who want to add logs for a specific `location`, for example, instead of the entire `http` block.

Has there been any discussion or thought around how we could let a user specify at that level to be more granular? I think it would probably end up being a policy or filter to do that, which would obviously be a lot more work than this. Since this work is just targeting logging at the `http` context, let's be clear in this comment that this only applies to that context. If we document this feature, we should also be clear about that.